.dotnet/src/Custom/Chat/ChatCompletionOptions.cs

```diff
@@ -159,4 +159,10 @@ public ChatCompletionOptions()
     // CUSTOM: Added the Obsolete attribute.
     [Obsolete($"This property is obsolete. Please use {nameof(Tools)} instead.")]
     public IList<ChatFunction> Functions { get; }
+
+    [CodeGenMember("Metadata")]
+    public IDictionary<string, string> Metadata { get; } = new ChangeTrackingDictionary<string, string>();
+
+    [CodeGenMember("Store")]
+    public bool? EnableOutputStorage { get; set; }
```

Comment:
nit: Is calling it `StoreOutput` better? Basically following the pattern that we used for `IncludeLogProbabilities` and `AllowParallelToolCalls` in this class.

Reply:
`AllowParallelToolCalls` is what I was attempting to pattern this off of; "Allow" just didn't feel like the right word since it's an explicitly directive "service, thou shalt store stuff" and not a "you can use parallel tool calls *if you want to*, model."

`StoreOutput` doesn't have a clear yes/no word in it for the boolean-ness; `Store` is confusable between being a noun and a verb on its own, and `Output` likewise has a lot of overloaded meanings. If were asked "what do you think a symbol named `StoreOutput` is?", my first guesses would be (a) a method; (b) a property providing an output from a store; and only then (c) a flag controlling whether output should be stored.